src/routers/MultiRouteRouter.js

```diff
@@ -35,16 +36,21 @@ class MultiRouteRouter extends ApplicationRouter<Array<Object>> {
     }
 
     // multiple destinations given
-    const routes = destinationNames.map((destinationName, index) => RouteUtils.getRoutes(
+    const multiRoutes = destinationNames.map((destinationName, index) => RouteUtils.getRoutes(
       destinationName,
       end[index],
       start,
       departureTimeQuery,
       false,
     ));
 
-    return Promise.all(routes).then(val => val).catch((err) => {
-      throw LogUtils.logErr(err, routes, 'Could not get all specified routes');
+    // return the best route for each different destination
+    return Promise.all(multiRoutes).then((val) => {
+      const bestRoutes = val.map(routes => ((routes.length > 0) ? routes[0] : null));
+      AnalyticsUtils.assignRouteIdsAndCache(bestRoutes.filter(r => r !== null));
```

Comment:
You can do `bestRoutes.filter(Boolean)` to filter out faulty values

Reply:
aren't non null checks `!=` and not `!==`?